src/pages/instances/actions/PublishInstanceBtn.tsx

```diff
@@ -0,0 +1,47 @@
+import { FC } from "react";
+import { LxdInstance } from "types/instance";
+import { ActionButton } from "@canonical/react-components";
+import usePortal from "react-useportal";
+import CreateImageFromInstanceForm from "../forms/CreateImageFromInstanceForm";
+
+interface Props {
+  instance: LxdInstance;
+  isDeleting: boolean;
+  isStopping: boolean;
+}
+
+const PublishInstanceBtn: FC<Props> = ({
+  instance,
+  isDeleting,
+  isStopping,
+}) => {
+  const { openPortal, closePortal, isOpen, Portal } = usePortal();
+
+  return (
+    <>
+      {isOpen && (
+        <Portal>
+          <CreateImageFromInstanceForm
+            close={closePortal}
+            instance={instance}
+          />
+        </Portal>
+      )}
+      <ActionButton
+        appearance="default"
+        onClick={openPortal}
+        aria-label="Publish instance"
+        title="Publish instance"
+        disabled={
+          instance?.status == ("Running" || "Restarting") ||
```

Comment:
did you test these statuses? From my testing, I can publish an instance while it's running, not sure about the other ones though. You should double check all statuses from `LxdInstanceStatus`, I think at least "Starting", "Stopping", "Restarting" and "Freezing" should result in the button being disabled, but please verify the other statuses as well.

You should also use `useInstanceLoading` to determine the status of an instance instead since it is managed via a global state provider.

Reply:
Yes, I did test them, and I cannot publish when running. I found all of the Instance statuses from another file as well but I will revisit this section and use useInstanceLoading.

Reply:
I think it would be a combination between `instance.status` and status returned by using `useInstanceLoading` (which represents instance transient states)

Reply:
![image](https://github.com/user-attachments/assets/110fbf7c-9682-40a2-aa2f-b339c0bd89b4)
In this file, RestartInstanceBtn, both useInstanceLoading and instance.status are used to determine whether an instance isLoading or not. Does that not mean that either application is valid?

Reply:
no because `instance.status` is fixed and its data will not change unless you make another API call. `useInstanceLoading` returns global states for the transient instance statuses i.e. `Starting`, `Freezing`, `Stopping` and `Restarting` which is dynamic. In fact I think the code is actually a bit wrong because those transient states does not actually exist for an instance in LXD, they exist only in the UI.

Reply:
Given you said that you believe the solution will be a combo between instance.status and useInstanceLoading, are you insinuating I should replicate the above code?

Reply:
yeah pretty much, use `instanceLoading` for the transient statuses comparison, then use `instance.status` for the lxd instance statuses comparison. I just tested publishing an instance using the CLI while it is running, it actually failed so I'm not sure what happened yesterday when I tried it in the UI. So I guess `Running` should result in the button being disabled as well.

Reply:
Nevermind, I think the new implementation I have addressed this.